arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,42 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 100;
+
+const int fSpeed   = 30; //70% of the full speed forward
+
+
+
+ArduinoRuntime arduinoRuntime;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+void drive()
+{
+car.setSpeed(fSpeed);
+car.setAngle(0);
+}
+
+void setup()
+{
+    Serial.begin(9600);
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+  if(front.getDistance() >= 100 || front.getDistance()==0)
```

Comment:
The problem with creating a new pull request, is that all previous conversations are lost. You can and should use `git` to resolve such issues.
Anyway, the problem with this remains. Refer to our previous conversation.